buildbot/master/files/config/master.cfg

```diff
@@ -145,49 +159,57 @@ def branch_priority(builder, requests):
 
 
-class DynamicServoBuilder(util.BuilderConfig):
+class DynamicRepoBuilder(util.BuilderConfig):
     """\
-    Builder which uses DynamicServoFactory to run steps
-    from a YAML file in the main servo repo.
+    Builder which uses DynamicGitUrlFactory to run steps
+    from a YAML file in the repo.
     """
-    def __init__(self, name, slavenames, environment):
+    def __init__(self, repo_owner, repo_name, name, slavenames, environment):
         # util.BuilderConfig is an old-style class so we cannot use super()
         # but must hardcode the superclass here
         util.BuilderConfig.__init__(
             self,
             name=name,
             slavenames=slavenames,
-            factory=factories.DynamicServoFactory(name, environment),
+            factory=factories.DynamicGitUrlFactory("https://github.com/%s/%s" % (repo_owner, repo_name),
+                                                   name, environment,
+                                                   "etc/ci/buildbot_steps.yml"),
             nextBuild=branch_priority,
             canStartBuild=util.enforceChosenSlave,
+            properties={
+                "repo_owner": repo_owner,
+                "repo_name": repo_name,
+            },
         )
 
 
 c['builders'] = [
-    DynamicServoBuilder("android", CROSS_SLAVES, envs.build_android),
-    DynamicServoBuilder("android-nightly", CROSS_SLAVES, envs.build_android),
-    DynamicServoBuilder("arm32", CROSS_SLAVES, envs.build_arm32),
-    DynamicServoBuilder("arm64", CROSS_SLAVES, envs.build_arm64),
-    DynamicServoBuilder("linux-dev", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("linux-nightly", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("linux-rel-css", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("linux-rel-intermittent", LINUX_SLAVES,
-                        envs.build_linux),
-    DynamicServoBuilder("linux-rel-nogate", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("linux-rel-wpt", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("mac-dev-unit", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-nightly", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-rel-css1", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-rel-css2", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-rel-intermittent", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-rel-wpt1", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-rel-wpt2", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-rel-wpt3", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("mac-rel-wpt4", MAC_SLAVES, envs.build_mac),
-    DynamicServoBuilder("windows-msvc-dev", WINDOWS_SLAVES,
-                        envs.build_windows_msvc),
-    DynamicServoBuilder("windows-msvc-nightly", WINDOWS_SLAVES,
-                        envs.build_windows_msvc),
+    DynamicRepoBuilder("servo", "servo", "android", CROSS_SLAVES, envs.build_android),
```

Comment:
It would be nice to keep `DynamicServoBuilder` as a subclass of `DynamicRepoBuilder` to cut down on a lot of this duplication, reduce diff noise, and also avoid hitting the 80-char line limit.